File: matrix/python/sol.py
Matrix = list[list[int]]

M = int(input("Rows: "))
N = int(input("Columns: "))

def setup_matrix(row: int, column: int) -> Matrix:
    a: Matrix = []
    for x in range(row):
        for y in range(column):
            if len(a) != x + 1:
                a.append([])
            if len(a[x]) != y + 1:
                a[x].append(0)
            a[x][y] = int(input(f"Enter number [{x + 1}][{y + 1}]: "))
    return a

print(f"Enter first matrix values ({M}x{N})")
first_matrix: Matrix = setup_matrix(M, N)
print(f"Enter second matrix values ({M}x{N})")
second_matrix: Matrix = setup_matrix(M, N)

result: Matrix = []

for x in range(M):
    for y in range(N):
        if len(result) != x + 1:
            result.append([])
        if len(result[x]) != y + 1:
            result[x].append(0)
        result[x][y] = first_matrix[x][y] + second_matrix[x][y]

print("Result")
for x in result:
    out = "     ".join(str(a) for a in x)
    print(out)

File: matrix/c/sol.c
/**
 * I hate C so much yet people using it
 */

#include <stdio.h>

void setup_matrix(int row, int col, int matrix[row][col]) {
    for (int i = 0; i < row; i++) {
        for (int j = 0; j < col; j++) {
            printf("Enter number [%d][%d]: ", i + 1, j + 1);
            scanf("%d", &matrix[i][j]);
        }
    }
}

int main() {
    int M, N;
    printf("Rows: ");
    scanf("%d", &M);
    printf("Columns: ");
    scanf("%d", &N);

    int first_matrix[M][N], second_matrix[M][N], result[M][N];

    printf("Enter first matrix values (%dx%d)\n", M, N);
    setup_matrix(M, N, first_matrix);

    printf("Enter second matrix values (%dx%d)\n", M, N);
    setup_matrix(M, N, second_matrix);

    for (int i = 0; i < M; i++) {
        for (int j = 0; j < N; j++) {
            result[i][j] = first_matrix[i][j] + second_matrix[i][j];
        }
    }

    printf("Result:\n");
    for (int i = 0; i < M; i++) {
        for (int j = 0; j < N; j++) {
            printf("%d\t", result[i][j]);
        }
        printf("\n");
    }

    return 0;
}
